src/components/UserSelector.tsx

```diff
@@ -1,33 +1,91 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { getUsers } from '../api/mate';
+
+interface Props {
+  selectedUser: User | null;
+  setSelectedUser: (user: User | null) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  setSelectedUser,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isOpend, setIsOpend] = useState(false);
+
+  const getUsersFromServer = useCallback(async () => {
+    const result = await getUsers();
+
+    setUsers(result);
+  }, []);
+
+  const handleToggleSelect = () => {
+    setIsOpend(prev => !prev);
+  };
+
+  const handleSelectUser = (user: User) => {
+    setSelectedUser(user);
+    setIsOpend(false);
+  };
+
+  useEffect(() => {
+    getUsersFromServer();
+  }, []);
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={classNames(
+        'dropdown',
+        { 'is-active': isOpend },
+      )}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={handleToggleSelect}
+          onBlur={() => setIsOpend(false)}
         >
-          <span>Choose a user</span>
+          <span>
+            {
+              selectedUser
+                ? selectedUser.name
+                : 'Choose a user'
+            }
+          </span>
 
           <span className="icon is-small">
             <i className="fas fa-angle-down" aria-hidden="true" />
           </span>
         </button>
       </div>
 
-      <div className="dropdown-menu" id="dropdown-menu" role="menu">
+      <div
+        className="dropdown-menu"
+        id="dropdown-menu"
+        role="menu"
+      >
         <div className="dropdown-content">
-          <a href="#user-1" className="dropdown-item">Leanne Graham</a>
-          <a href="#user-2" className="dropdown-item is-active">Ervin Howell</a>
-          <a href="#user-3" className="dropdown-item">Clementine Bauch</a>
-          <a href="#user-4" className="dropdown-item">Patricia Lebsack</a>
-          <a href="#user-5" className="dropdown-item">Chelsey Dietrich</a>
+          {users.map(user => (
+            <a
```

Comment:
When you have free time please read this [interesting article](https://medium.com/@h_locke/links-vs-buttons-e8b523660fb3) or any similar to it when searching in google `button vs link philosophy`